extra/src/main/java/aima/extra/probability/constructs/ProbabilityComputation.java

```diff
@@ -0,0 +1,141 @@
+package aima.extra.probability.constructs;
+
+import java.math.BigInteger;
+import java.math.MathContext;
+import aima.extra.probability.ProbabilityNumber;
+
+/**
+ * ProbabilityComputation is a helper class which allows for ProbabilityNumber
+ * operations to return ProbabilityNumber objects that may contain invalid
+ * values. It is possible for intermediate calculations to result in invalid
+ * probability values, whereas the outcome after applying all the operations
+ * results in a valid probability value.
+ * 
+ * @author Nagaraj Poti
+ */
+public class ProbabilityComputation {
```

Comment:
Remove checkrequired logic as it makes things complicated and introduces side effects and potential concurrency issues in multi-threaded environments. 

Update class javadoc to indicate use of this class in case of streams as this is the primary intent of this class.